int32 ad_x_error
int32 ad_y_error
int32 ad_proximity
